statsmodels/tools/sequences.py

```diff
@@ -0,0 +1,185 @@
+
+    Returns
+    -------
+    primes : list(int)
+        List of primes.
+
+    """
+    primes = [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59,
+              61, 67, 71, 73, 79, 83, 89, 97, 101, 103, 107, 109, 113, 127,
+              131, 137, 139, 149, 151, 157, 163, 167, 173, 179, 181, 191, 193,
+              197, 199, 211, 223, 227, 229, 233, 239, 241, 251, 257, 263, 269,
+              271, 277, 281, 283, 293, 307, 311, 313, 317, 331, 337, 347, 349,
+              353, 359, 367, 373, 379, 383, 389, 397, 401, 409, 419, 421, 431,
+              433, 439, 443, 449, 457, 461, 463, 467, 479, 487, 491, 499, 503,
+              509, 521, 523, 541, 547, 557, 563, 569, 571, 577, 587, 593, 599,
+              601, 607, 613, 617, 619, 631, 641, 643, 647, 653, 659, 661, 673,
+              677, 683, 691, 701, 709, 719, 727, 733, 739, 743, 751, 757, 761,
+              769, 773, 787, 797, 809, 811, 821, 823, 827, 829, 839, 853, 857,
+              859, 863, 877, 881, 883, 887, 907, 911, 919, 929, 937, 941, 947,
+              953, 967, 971, 977, 983, 991, 997][:n]
+
+    if len(primes) < n:
+        big_number = 10
+        while 'Not enought primes':
+            primes = primes_from_2_to(big_number)[:n]
+            if len(primes) == n:
+                break
+            big_number += 1000
+
+    return primes
+
+
+def van_der_corput(n_sample, base=2):
+    """Van der Corput sequence.
+
+    Parameters
+    ----------
+    n_sample : int
+        Number of element of the sequence.
+    base : int
+        Base of the sequence.
+
+    Returns
+    -------
+    sequence : list (n_samples,)
+        Sequence of Van der Corput.
+
+    """
+    sequence = []
+    for i in range(n_sample):
+        n_th_number, denom = 0., 1.
+        quotient = i
+        while quotient > 0:
+            quotient, remainder = divmod(quotient, base)
+            denom *= base
+            n_th_number += remainder / denom
+        sequence.append(n_th_number)
+
+    return sequence
+
+
+def halton(dim, n_sample, feature_range=(0, 1)):
+    """Halton sequence.
+
+    Parameters
+    ----------
+    dim : int
+        Dimension of the parameter space.
+    n_sample : int
+        Number of samples to generate in the parametr space.
+    feature_range : tuple or array_like ([min, n_features], [max, n_features])
+        Desired range of transformed data.
+
+    Returns
+    -------
+    sequence : array_like (n_samples, n_features)
+        Sequence of Halton.
+
+    References
+    ----------
+    [1] Halton, "On the efficiency of certain quasi-random sequences of points
+      in evaluating multi-dimensional integrals", Numerische Mathematik, 1960.
+
+    """
+    base = n_primes(dim)
+
+    # Generate a sample using a Van der Corput sequence per dimension.
+    sample = [van_der_corput(n_sample + 1, dim) for dim in base]
+    sample = np.stack(sample, axis=-1)[1:]
+
+    # Sample scaling from unit hypercube to feature range
```

Comment:
I would make this optional, with no rescaling. AFAIU the default should be in the interior of the unit cube, is it?
Eg. for unbound transformation, e.g. for normal distribution we don't want to include the boundaries, i.e. keep open unit cube.

Reply:
I have made this optional.

Yes, a Halton sequence returns points in the unit hypercube (bounded). But this is not an issue as we can still apply an inverse CDF to match any unbounded distribution.